File: mycollab-web/src/main/java/com/esofthead/mycollab/module/project/view/parameters/ProjectMemberScreenData.java
package com.esofthead.mycollab.module.project.view.parameters;

import com.esofthead.mycollab.module.project.domain.ProjectMember;
import com.esofthead.mycollab.module.project.domain.criteria.ProjectMemberSearchCriteria;
import com.esofthead.mycollab.vaadin.mvp.ScreenData;

public class ProjectMemberScreenData {
	public static class Search extends ScreenData<ProjectMemberSearchCriteria> {

		public Search(ProjectMemberSearchCriteria params) {
			super(params);
		}
	}

	public static class Add extends ScreenData<ProjectMember> {

		public Add(ProjectMember params) {
			super(params);
		}
	}

	public static class InviteProjectMembers extends ScreenData {

		public InviteProjectMembers() {
			super(null);
		}
	}

	public static class Read extends ScreenData<Object> {

		public Read(Object params) {
			super(params);
		}
	}
}

File: mycollab-web/src/main/scala/com.esofthead.mycollab.module.project.view.parameters/ProjectMemberScreenData.scala
package com.esofthead.mycollab.module.project.view.parameters

import com.esofthead.mycollab.module.project.domain.ProjectMember
import com.esofthead.mycollab.module.project.domain.criteria.ProjectMemberSearchCriteria
import com.esofthead.mycollab.vaadin.mvp.ScreenData

/**
 * @author MyCollab Ltd.
 * @since 5.0.3
 */
object ProjectMemberScreenData {

  class Search(params:ProjectMemberSearchCriteria) extends ScreenData[ProjectMemberSearchCriteria](params) {}

  class Add(params:ProjectMember) extends ScreenData[ProjectMember](params){}

  class InviteProjectMembers extends ScreenData {}

  class Read(params:Object) extends ScreenData[Object](params) {}
}
